packages/alchemy/src/middleware/gasManager.ts

```diff
@@ -123,15 +123,9 @@ export function alchemyGasManagerMiddleware<C extends ClientWithAlchemyMethods>(
     gasEstimator: disableGasEstimation
       ? fallbackGasEstimator
       : async (struct, { overrides, account, feeOptions }) => {
-          const entryPoint = account.getEntryPoint();
           // if user is bypassing paymaster to fallback to having the account to pay the gas (one-off override),
           // we cannot delegate gas estimation to the bundler because paymaster middleware will not be called
-          if (
-            (entryPoint.version === "0.6.0"
-              ? (overrides as UserOperationOverrides<"0.6.0">)?.paymasterAndData
-              : (overrides as UserOperationOverrides<"0.7.0">)
-                  ?.paymasterData) === "0x"
-          ) {
+          if (overrides && bypassPaymasterAndData(overrides)) {
```

Comment:
Correct me if I'm wrong, but since `bypassPaymasterAndData` accepts `undefined`, you don't need the `overrides &&` here, likewise below.